TestProject/UnitTest1.cs

```diff
@@ -1,10 +1,29 @@
-namespace TestProject;
+namespace Training.EState;
 
 [TestClass]
-public class UnitTest1
-{
-    [TestMethod]
-    public void TestMethod1()
-    {
-    }
+public class UnitTest1 {
+
+   [TestMethod]
+   public void FileParseTest () {
+      Assert.AreEqual (("C", "PROGRAM", "DATA", "EXT"), Program.GetParsedFilePath ("C:\\PROGRAM\\DATA.EXT"));
+      Assert.AreEqual (("W", "TRAINING", "PROGRAM", "CS"), Program.GetParsedFilePath ("W:\\TRAINING\\PROGRAM.CS"));
+      Assert.AreEqual (("C", "USERS\\DOWNLOADS", "ASSIGNMENT", "DOCX"), Program.GetParsedFilePath ("C:\\USERS\\DOWNLOADS\\ASSIGNMENT.DOCX"));
+      Assert.AreEqual (("C", "PROGRAM", "DATA", "EXT"), Program.GetParsedFilePath ("C:/PROGRAM/DATA.EXT"));
+      Assert.AreEqual (("W", "TRAINING", "PROGRAM", "CS"), Program.GetParsedFilePath ("W:/TRAINING\\PROGRAM.CS"));
+      Assert.AreEqual (("W", "TRAINING", "PROGRAM", "CS"), Program.GetParsedFilePath ("W:\\TRAINING/PROGRAM.CS"));
+   }
+
+   [TestMethod]
+   public void InValidFileParser () {
+      Assert.ThrowsException<InValidPathException> (() => Program.GetParsedFilePath ("C:\\DATA.EXT"));

```

Comment:
As we are appending ~ at end of file path. Use ~ at the end in one of test case.